tests/test_integration.py

```diff
@@ -378,3 +378,362 @@ def test(self, soco):
+        que = soco.get_queue()[1:]
+        for s_item, q_item in zip(spl, que):
+            assert s_item.resources[0].uri == q_item.resources[0].uri
+
+    def test_remove_first_track_full(self, soco):
+        """ Test removing first track from Sonos Playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        tracks = [0] + range(num_tracks - 1)    # [0, 0, 1, ..., n-1]
+        new_pos = [None, ] + range(num_tracks - 1) #[None, 0, ..., n-1]
+        args = {'object_id': test_pl.item_id,
+                'tracks': tracks,
+                'new_pos': new_pos}
+        change, count, update_id = soco.reorder_sonos_playlist(**args)
+        assert change == -1
+        assert count == num_tracks - 1
+        assert update_id != 0
+        spl = soco.browse(ml_item=test_pl)
+        que = soco.get_queue()[1:]
+        for s_item, q_item in zip(spl, que):
+            assert s_item.resources[0].uri == q_item.resources[0].uri
+
+    def test_remove_last_track(self, soco):
+        """ Test removing last track from Sonos Playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        tracks = range(num_tracks)
+        new_pos = range(num_tracks - 1) + [None, ]
+        args = {'object_id': test_pl.item_id,
+                'tracks': tracks,
+                'new_pos': new_pos}
+        change, count, update_id = soco.reorder_sonos_playlist(**args)
+        assert change == -1
+        assert count == num_tracks - 1
+        assert update_id != 0
+        spl = soco.browse(ml_item=test_pl)
+        que = soco.get_queue()[:-1]
+        for s_item, q_item in zip(spl, que):
+            assert s_item.resources[0].uri == q_item.resources[0].uri
+
+    def test_remove_between_track(self, soco):
+        """ Test removing a middle track from Sonos Playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        ndx = int(num_tracks / 2)
+        tracks = [ndx]
+        new_pos = [None]
+        args = {'object_id': test_pl.item_id,
+                'tracks': tracks,
+                'new_pos': new_pos}
+        change, count, update_id = soco.reorder_sonos_playlist(**args)
+        assert change == -1
+        assert count == num_tracks - 1
+        assert update_id != 0
+        spl = soco.browse(ml_item=test_pl)
+        que = soco.get_queue()
+        del que[ndx]
+        for s_item, q_item in zip(spl, que):
+            assert s_item.resources[0].uri == q_item.resources[0].uri
+
+    def test_remove_some_tracks(self, soco):
+        """ Test removing some tracks from Sonos Playlist. """
+        test_pl = self._reset_spl_contents(soco)
+        num_tracks = soco.browse(ml_item=test_pl).total_matches
+        # get rid of the even numbered tracks
+        tracks = sorted([x for x in range(num_tracks) if not x & 1],
```

Comment:
I'd prefer the form without bitwise operators for something like this: if x % 2 == 1